### ISSUE TITLE
Integrating some pooling methods from the miceafter and marginaleffects packages

### ISSUE THREAD

**Issue by Generalized** [insightsengineering/rbmi#387]
In one of the vignettes I noticed an example of custom analysis using the glm() to deal with categorical variables.

It's common to also analyze binary data. It can be done by either directly using the formulas for CI for the difference in proportions (Wald's, Miettinen-Nurminen, etc) or (for Wald) - by using the logistic regression followed by the analysis of marginal effects (to get the %point change rather than OR), along with appropriate CI. This requires own pooling methods. Sometimes we also need to calculate some planned non-parametric tests, so we end up with D2 approach (using the Rubin's terminology used in the "mice" family).

Let me show a trivial example with 2 arms and binary outcome. I compare the % of successes and calculate the Wald's CI for the purpose of the further non-inferiority analysis. First I do it via logistic regression + average marginal effects, then by using the direct formula for the CI of the difference. They are almost identical, with the difference, that the GLM uses z (normal distribution with infinite DoF) and the direct formula uses the t distribution.

```{r}
library(mice)
library(marginaleffects)
library(miceafter)

fit_logistic <- function(dat) {
  mod <- glm(endp ~ trt, family = binomial(link = 'logit'), data=dat)
  out <- avg_slopes(mod, newdata = dat)
  return(out)
}

data_miss <- structure(list(id = 1:37, trt = c("soc", "soc", "soc", "soc", 
"soc", "soc", "soc", "soc", "soc", "soc", "soc", "soc", "soc", 
"soc", "soc", "soc", "soc", "soc", "soc", "soc", "soc", "arm", 
"arm", "arm", "arm", "arm", "arm", "arm", "arm", "arm", "arm", 
"arm", "arm", "arm", "arm", "arm", "arm"), endp = structure(c(NA, 
1L, 2L, 1L, NA, 1L, NA, 1L, 2L, 1L, 2L, 1L, 2L, 1L, 2L, 2L, 2L, 
2L, 2L, 2L, 2L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 2L, NA, 2L, 2L, 2L, 
1L, 1L, 1L, 1L), levels = c("TRUE", "FALSE"), class = "factor")), row.names = c(NA, 
-37L), class = "data.frame")

set.seed(1000)
imp <- mice::mice(data_miss, m = 20, method = "pmm",  maxit = 50, seed = 1000, print = FALSE)
fit <- with(imp,  glm(endp ~ trt, family = binomial(link = 'logit')))
summary(pool(fit), conf.int = TRUE, exponentiate = TRUE)

1 (Intercept) 0.3658376 0.5856050 -1.717140 30.78614 0.09599953 0.1107753  1.208186
2      trtsoc 3.5810170 0.7504718  1.699793 30.03251 0.09950603 0.7734058 16.580795

dat_mice <- complete(imp, "all")
mod_imputation <- lapply(dat_mice, fit_logistic)
summary(pool(mod_imputation), conf.int = TRUE)

 term  contrast  estimate std.error statistic       df    p.value      2.5 %    97.5 %
1  trt soc - arm 0.2979167  0.162139  1.837415 2208.092 0.06628299 -0.0200443 0.6158776

#---------- simple Wald's diff pooled

imp_dat <- mids2milist(imp)
ra <- with(imp_dat, expr=propdiff_wald(as.numeric(2-as.numeric(factor(endp))) ~ as.numeric(2-as.numeric(factor(trt)))))
pool_propdiff_wald(ra)

     Prop diff Wald      SE       t 95 CI low 95 CI high
[1,]        0.29792 0.16214 2.04071  -0.03296    0.62879
attr(,"class")
[1] "mipool"
```

In both cases appropriate pooling methods were used. 
The miceafter (and miceadds) + marginaleffects packages offer a number of methods of pooling the results for several kinds of analyses/

Since you allow for custom analyses (like in the example employing glm), would you consider adding some support for those packages **in terms of pooling**? Either by copying the pooling formulas from them, or by adding "interface" that could "mimic" the one from mice, that these functions can understand?

In other words - to make one able to conduct the above simple analysis using the RBMI approach.

**Comment by danielinteractive** [comment]
Thanks @Generalized for reaching out on this! @wolbersm @gowerc what do you think?

**Comment by wolbersm** [comment]
Hi @Generalized 
Thanks! We haven't thought of this yet. 

1) The statistical method for pooling in rbmi depends on the imputation method (specified in `draws()`) and the methods implements in mice (or add-on packages) would only be applicable to `method_bayes()` and `method_approxbayes()`. 

2) For `method_bayes()` and `method_approxbayes()` it should be relatively straightforward to extract the imputed datasets (using the `extract_imputed_dfs()` helper function) and then put them in a format which allows pooling/post-processing via mice-tools. Could you try this @Generalized? 
In the mid-term, I think this would be a nice-to-have feature and it would be useful to have a short example/vignette/function which illustrates how this is done. @gowerc and @nociale what do you think?

**Comment by wolbersm** [comment]
Hi @gowerc 
Thanks a lot, very nice! To me this is enough illustration on how to "hand-over" `rbmi` results to `mice` and I hope this solves the question of @Generalized .  I don't think more formal integration of this is mandated for now.
